dds/DCPS/DataWriterImpl_T.h

```diff
@@ -417,6 +417,8 @@ DataWriterImpl_T
 
 private:
 
+  static const int PADDING_MARKER_BYTE_INDEX = 3;
```

Comment:
Move to Serializer?

This could keep more encoding details out of DataWriterImpl_T.  How about passing the message block itself to `EncapsulationHeader::set_encapsulation_options`?  The `rd_ptr` of the message block already points to the start of the block.  The `length` of the message block is number of bytes actually used.  That is enough to determine the options.

Reply:
Yes I agree. Technically we should be doing this everything a encapsulation header is being used, so that would make it easier to set the padding marker in other places if we wished.